modules/cactus-web/src/BrandBar/BrandBar.tsx

```diff
@@ -205,6 +290,190 @@ const StyledUserMenu = styled.div`
   outline: none;
 `
 
+const Dropdown: React.FC<DropdownProps> = ({ label, children }) => {
+  const buttonRef = React.useRef<HTMLButtonElement>(null)
+  const portalRef = React.useRef<HTMLDivElement>(null)
+  const isTiny = SIZES.tiny === useScreenSize()
+  const { expanded, toggle, buttonProps, popupProps, setFocus } = usePopup('dialog', {})
+
+  React.useEffect(() => {
+    const handleBodyClick = (event: MouseEvent): void => {
+      const { target } = event
+      if (
+        target instanceof Node &&
+        !buttonRef.current?.contains(target) &&
+        !portalRef.current?.contains(target)
+      ) {
+        toggle(false)
+      }
+    }
+
+    const handleBodyKeyDown = (event: KeyboardEvent) => {
+      const { target } = event
+      if (
+        target instanceof Node &&
+        !buttonRef.current?.contains(target) &&
+        !portalRef.current?.contains(target)
+      ) {
+        toggle(false)
+      }
+    }
+
+    document.body.addEventListener('click', handleBodyClick)
+    document.body.addEventListener('keydown', handleBodyKeyDown)
+
+    return () => {
+      document.body.removeEventListener('click', handleBodyClick)
+      document.body.removeEventListener('keydown', handleBodyKeyDown)
+    }
+  })
```

Comment:
Pretty sure if you omit the dependency array entirely it runs the effect on every render, which is probably not what you want here.
```suggestion
  }, [])
```

Reply:
Whoops, yeah that definitely wasn't intentional.